packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -39,21 +40,339 @@ import {
+      param2 instanceof PrimitiveInstanceValue
+    ) {
+      if (
+        param1.genericType.value.rawType.name === PRIMITIVE_TYPE.LATESTDATE &&
+        param2.genericType.value.rawType.name === PRIMITIVE_TYPE.LATESTDATE
+      ) {
+        return true;
+      } else if (
+        param1.genericType.value.rawType.name ===
+        param2.genericType.value.rawType.name
+      ) {
+        return param1.values[0] === param2.values[0];
+      }
+      return false;
+    }
+    return false;
+  };
+  switch (stereotype) {
+    case MILESTONING_STEREOTYPE.BITEMPORAL:
+      if (
+        idx === 0 &&
+        checkIfEquivalent(milestoningParameter, querySetupState.processingDate)
+      ) {
+        return true;
+      } else if (
+        idx === 1 &&
+        checkIfEquivalent(milestoningParameter, querySetupState.businessDate)
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    case MILESTONING_STEREOTYPE.PROCESSING_TEMPORAL:
+      if (
+        checkIfEquivalent(milestoningParameter, querySetupState.processingDate)
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    case MILESTONING_STEREOTYPE.BUSINESS_TEMPORAL:
+      if (
+        checkIfEquivalent(milestoningParameter, querySetupState.businessDate)
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    default:
+  }
+  return false;
+};
+
+export const isDefaultDatePropagationSupported = (
+  currentPropertyExpression: AbstractPropertyExpression,
+  queryBuilderState: QueryBuilderState,
+  prevPropertyExpression?: AbstractPropertyExpression | undefined,
+): boolean => {
+  const property = currentPropertyExpression.func;
+  const graph = queryBuilderState.graphManagerState.graph;
+  if (
+    prevPropertyExpression &&
+    prevPropertyExpression.func.genericType.value.rawType instanceof Class
+  ) {
+    const milestoningStereotype = getMilestoneTemporalStereotype(
+      prevPropertyExpression.func.genericType.value.rawType,
+      graph,
+    );
+    if (
+      milestoningStereotype &&
+      !prevPropertyExpression.parametersValues
+        .slice(1)
+        .every(
+          (parameterValue) =>
+            parameterValue instanceof INTERNAL__PropagatedValue,
+        )
+    ) {
+      return false;
+    }
+  }
+  if (property.genericType.value.rawType instanceof Class) {
+    // sourceStereotype: Stereotype of source class of current property expression.
+    const sourceStereotype =
+      property instanceof DerivedProperty
+        ? getDerivedPropertyMilestoningSteoreotype(property, graph)
+        : undefined;
+    if (sourceStereotype === MILESTONING_STEREOTYPE.BITEMPORAL) {
+      return true;
+    }
+
+    // targetStereotype: Stereotype (if exists) of genericType of current property expression.
+    const targetStereotype = getMilestoneTemporalStereotype(
+      property.genericType.value.rawType,
+      graph,
+    );
+    return sourceStereotype === targetStereotype;
```

Comment:
I think there might a bug here also since we could have `sourceStereotype === undefined && targetStereotype === undefined` and that returns `true`